Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/TruthSeedingAlgorithm.hpp

```diff
@@ -0,0 +1,78 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/Definitions/Units.hpp"
+#include "ActsExamples/EventData/ProtoTrack.hpp"
+#include "ActsExamples/EventData/SimSeed.hpp"
+#include "ActsExamples/Framework/BareAlgorithm.hpp"
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Acts {
+class TrackingGeometry;
+}
+
+namespace ActsExamples {
+
+/// Construct track seeds from .
+class TruthSeedingAlgorithm final : public BareAlgorithm {
+ public:
+  struct Config {
+    /// The input truth particles that should be used for truth seeding.
+    std::string inputParticles;
+    /// The input hit-particles map collection.
+    std::string inputMeasurementParticlesMap;
+    /// Input source links collection.
+    std::string inputSourceLinks;
+    /// Input space point collections.
+    ///
+    /// We allow multiple space point collections to allow different parts of
+    /// the detector to use different algorithms for space point construction,
+    /// e.g. single-hit space points for pixel-like detectors or double-hit
+    /// space points for strip-like detectors.
+    std::vector<std::string> inputSpacePoints;
+    /// Output successfully seeded truth particles.
+    std::string outputParticles;
+    /// Output full proto track collection.
+    std::string outputFullProtoTracks;
+    /// Output seed collection.
+    std::string outputSeeds;
+    /// Output proto track collection.
+    std::string outputProtoTracks;
```

Comment:
Here I'm not sure about the naming... I would rather go for something like: 
* `outputFullProtoTracks` -> `outputProtoTracks`
* `outputProtoTracks` -> `output3HitProtoTrack` / `outputThreeHitProtoTrack`

My natural assumption would be that a standard proto-track is a whole track. This also makes more explicit what in the short one is.

Reply:
I thought the same first but then I realized that our other seeders also use `prototracks` to communicate the 3 hits which went into the estimate

that said - not sure what the intention behind `prototracks`. maybe using it for truth tracking is the exception here?

/cc @paulgessinger @asalzburger @LuisFelipeCoelho

Reply:
I think it's fine to use ProtoTracks for seeds as well if they can serve the needs, it's probably just a special case for the truth tracking / refitting, I would say